src/Microsoft.IdentityModel.Tokens.Saml/SamlTokenUtilities.cs

```diff
@@ -102,5 +100,62 @@ internal static IEnumerable<SecurityKey> GetKeysForTokenSignatureValidation(stri
             }
             return null;
         }
+
+        /// <summary>
+        /// Creates <see cref="Claim"/>'s from <paramref name="claimsCollection"/>.
+        /// </summary>
+        /// <param name="claimsCollection"> A dictionary that represents a set of claims.</param>
+        /// <returns> A collection of <see cref="Claim"/>'s created from the <paramref name="claimsCollection"/>.</returns>
+        internal static IEnumerable<Claim> CreateClaimsFromDictionary(IDictionary<string, object> claimsCollection)
+        {
+            List<Claim> claims = null;
+            if (claimsCollection == null)
+                return claims;
+
+            claims = new List<Claim>();
+            foreach (var claim in claimsCollection)
+            {
+                string claimType = claim.Key;
+                object claimValue = claim.Value;
+
+                if (claimValue != null)
+                {
+                    string valueType = TokenUtilities.GetClaimValueTypeFromValue(claimValue);
+
+                    if (claimValue.GetType().Name == typeof(List<>).Name)
+                    {
+                        foreach (var item in (IList)claimValue)
+                            claims.Add(new Claim(claimType, item.ToString(), valueType));
```

Comment:
assume you would have to get the value type for each of the claims?

Reply:
I think we need to get valueType for every item in list and GetClaimValueTypeFromValue returns value.GetType().ToString() if a List contains any object types that are not covered. I will make that change.

Reply:
@RojaEnnam 
I think we should support 1 level of enumeration for a single item and throw NotSupported for lists of lists (and other collections).

Reply:
@brentschmaltz 
For second level of enumeration should we throw Exception or continue by assigning value.GetType() as valueType?

Reply:
After looking into it a bit more. I think we should throw NotSupported